spirv_msl.cpp

```diff
@@ -6549,6 +6673,58 @@ void CompilerMSL::prepare_access_chain_for_scalar_access(std::string &expr, cons
 	}
 }
 
+// Sets the interface member index for an access chain to a pull-model interpolant.
+void CompilerMSL::emit_interpolant_access_chain(const uint32_t *ops, uint32_t length)
+{
+	auto *var = maybe_get_backing_variable(ops[2]);
+	if (!var || !pull_model_inputs.count(var->self))
+		return;
+	// Get the base index.
+	uint32_t interface_index;
+	auto &var_type = get_variable_data_type(*var);
+	auto &result_type = get<SPIRType>(ops[0]);
+	auto *type = &var_type;
+	if (var_type.basetype == SPIRType::Struct)
```

Comment:
I think this will break if apps do partial access chains. E.g. OpAccessChain a little bit, and then continue the access chain based on the first one.

Reply:
Not anymore, it shouldn't.